### ISSUE TITLE
valid URL containing an underscore cannot be added

### ISSUE THREAD

**Issue by gingerbeardman** [sissbruecker/linkding#206]
From #202 

Confirming a valid URL containing an underscore cannot be added to the system

<img width="671" alt="Screenshot 2022-02-22 at 16 49 08" src="https://user-images.githubusercontent.com/49612/155180578-2ecfd4df-9373-4c4e-aea7-520eb1c63da2.png">

Example URLs
- http://infoseek_rip.g.ribbon.to/crspalm.hp.infoseek.co.jp/
- http://a_pollett.tripod.com/mayumihf.htm

So there is a problem with the process of checking a URLs, as both of the above are valid.

**Comment by sissbruecker** [comment]
Yeah, URL validation is not perfect, and from what I understand this is quite opinionated. For now linkding just uses the default Djanog URL validator, so you could also report it there. As a workaround linkding provides an option to disable URL validation: https://github.com/sissbruecker/linkding/blob/master/docs/Options.md#ld_disable_url_validation

**Comment by gingerbeardman** [comment]
Django won't fix

https://code.djangoproject.com/ticket/18517

**Comment by gingerbeardman** [comment]
I'm here one year later trying linkding again (thanks to Browser Companion app for iOS/macOS) and these two URLs failed to import again, not that I expected it to have changed.

:)

**Comment by gingerbeardman** [comment]
Here's another one:
- https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/20.04/x86_64/

**Comment by webysther** [comment]
Switch to use https://github.com/python-validators/validators/ can be the trick, support very well RFC and can be used to expand in future to support another types of url like custom schema and local uri using feature-flag.

**Comment by webysther** [comment]
> I'm here one year later trying linkding again (thanks to [Browser Companion](https://github.com/acez/bookmark-companion) app for iOS/macOS) and these two URLs failed to import again, not that I expected it to have changed.
> 
> ~I'm trying a workaround encoding the underscore as %5F~
> 
> I was able to add them as wayback machine versions.
> 
> eg
> 
> https://web.archive.org/web/20230504175240/http://infoseek_rip.g.ribbon.to/crspalm.hp.infoseek.co.jp/
> 
> 😄

Why not disable the validation to keep using it?

**Comment by gingerbeardman** [comment]
I simply do not want to turn off validation, I might change my mind in future.

**Comment by Mansarde** [comment]
I just tried to disable validation using `LD_DISABLE_URL_VALIDATION` set to false, but Linkding still validates URLs. :-\

**Comment by 0x1fff** [comment]
Another example of proper url which is not passing validation: 

- http://changy-.github.io/articles/xiao-mi-band-protocol-analyze.html
